core/Enclave/enclave_hsm.cpp

```diff
@@ -208,11 +344,11 @@ static sgx_status_t sgx_gcm_decrypt(const sgx_aes_gcm_128bit_key_t *key,
     return ret;
 }
 
-
-sgx_status_t enclave_create_aes_key(uint8_t *cmk_blob, uint32_t cmk_blob_size, uint32_t *req_blob_size)
+sgx_status_t enclave_create_aes_key(uint8_t *cmk_blob, uint32_t cmk_blob_size, uint32_t *req_blob_size, ehsm_keyspec_t keyspec)
 {
     sgx_status_t ret = SGX_ERROR_UNEXPECTED;
-    uint32_t real_blob_len = sgx_calc_gcm_data_size(0, SGX_AES_KEY_SIZE);
+    uint32_t key_size = sgx_get_key_size(keyspec);
+    uint32_t real_blob_len = sgx_calc_gcm_data_size(0, key_size);
 
     if (real_blob_len == UINT32_MAX)
```

Comment:
if (real_blob_len == UINT32_MAX ||  key_size == UINT32_MAX)

Reply:
I'd like to check the value of key_size before init real_blob_len.